conjure-http/src/client.rs

```diff
@@ -53,6 +55,37 @@ pub trait Client {
         U: VisitResponse<Self::BinaryBody>;
 }
 
+/// A trait implemented by async HTTP client implementations.
+pub trait AsyncClient {
```

Comment:
There's a rustc bug that prevents us from using `#[async_trait]` for this, but that's not a huge deal since we don't expect many things to actually have to implement this. https://github.com/rust-lang/rust/issues/63033

Reply:
same thing true for https://github.com/palantir/conjure-rust/pull/85/files#diff-adeb12920f549fa245147606171e1adeR45 ?

Reply:
All implementations of that should be generated by conjure-codegen, so there's no real need to make it ergonomic.